pkgs/pkgs-lib/formats.nix

```diff
@@ -274,6 +274,68 @@ rec {
 
   };
 
+  # Format for defining configuration of some PHP services, that use "include 'config.php';" approach.
+  php = {finalVariable}: with lib; let
+    toPHP = value: with builtins;
+      if value == null then "null" else
+      if value == true then "true" else
+      if value == false then "false" else
+      if isInt value || isFloat value then toString value else
+      if isString value then string value else
+      if isAttrs value then attrs value else
+      if isList value then list value else
+      abort "formats.php: should never happen (value = ${value})";
+
+    escapePHP = escape [ "\\" "\"" ];
+    string = value: "\"${escapePHP value}\"";
```

Comment:
Why a double quote? This means that `${foo}`, `{$foo}` and `$foo` are substitutable.

Reply:
Oops, I didn't know that. Thanks, I will fix.